### PR TITLE
Add SCL support to cassandra::schema

### PR DESCRIPTION
**Pull request by russellshackleford**
#### Pull Request (PR) description

Enable running `cqlsh` via python from software collections (SCL). Useful for RHEL/CentOS 6.

#### This Pull Request (PR) fixes the following issues

N/A

#### Additional info

The commits are all separated for easier review. Documentation will be done pending approval of the code changes.

### CODE REVIEW

**Comment by dallinb** [comment]
Looks good to me and a good suggestion to support SCL.

File: manifests/schema.pp [hunk 1]
```diff
@@ -49,9 +49,27 @@
   $permissions              = {},
   $tables                   = {},
   $users                    = {},
+  $use_scl                  = $::cassandra::params::use_scl,
```
Comment (bastelfreak) [inline manifests/schema.pp]: please don't add `::` after `$`, that is a deprecated syntax that was required for Puppet 3.

File: manifests/schema.pp [hunk 2]
```diff
@@ -49,9 +49,27 @@
...
+  $scl_name                 = $::cassandra::params::scl_name,
   ) inherits cassandra::params {
   require '::cassandra'
 
+  # Test that the SCL is valid
+  if $use_scl {
+    exec { "test ${scl_name} SCL":
+      command => "/usr/bin/scl enable ${scl_name} \"echo test\"",
```
Comment (bastelfreak) [inline manifests/schema.pp]: will the exec be executed during each puppet run? I don't think that this is optimal or even required?

File: manifests/schema/cql_type.pp [hunk 3]
```diff
@@ -16,23 +16,47 @@
   $ensure = present,
   $fields = {},
   $cql_type_name = $title,
+  $use_scl       = $::cassandra::params::use_scl,
```
Comment (bastelfreak) [inline manifests/schema/cql_type.pp]: please remove the `::`

File: manifests/schema.pp (hunk 2 above)
Comment (russellshackleford) [reply manifests/schema.pp]: The exec would indeed have run every time. I have removed this.

File: manifests/schema.pp (hunk 1 above)
Comment (russellshackleford) [reply manifests/schema.pp]: Removed

File: manifests/schema/cql_type.pp (hunk 3 above)
Comment (russellshackleford) [reply manifests/schema/cql_type.pp]: Removed :: and added datatype

**Comment by bastelfreak** [comment]
thanks for the work!

**Pull request merged**